src/armonik_analytics/metrics/time_series.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+
+import numpy as np
+
+from .base import ArmoniKMetric
+from armonik.common import Task
+
+
+class TasksInStatusOverTime(ArmoniKMetric):
+    """
+    A metric to track tasks in a particular status over time.
+    """
+
+    def __init__(self, timestamp, next_timestamp=None) -> None:
+        """
+        Initialize the metric.
+
+        Args:
+            timestamp (str): The current timestamp of the tasks.
+            next_timestamp (str, optional): The next timestamp of the tasks. Defaults to None.
+        """
+        self.timestamp = timestamp
+        self.next_timestamp = next_timestamp
+        self.timestamps = None
+        self.index = 0
+
+    def update(self, total: int, tasks: list[Task]) -> None:
+        """
+        Update the metric.
+
+        Args:
+            total (int): Total number of tasks.
+            tasks (list[Task]): A task batch.
+        """
+        n_tasks = len(tasks)
+        if self.timestamps is None:
+            n = total * 2 + 1 if self.next_timestamp else total + 1
```

Comment:
```suggestion
            n = (2 * total) + 1 if self.next_timestamp else total + 1
```

to improve the lisibility (some people forgot the rules)